processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -34,10 +35,66 @@
 
 /**
  */
-public abstract class BaseQuery<T> implements Query<T>
+public abstract class BaseQuery<T extends Comparable<T>> implements Query<T>
 {
+  public static <T> int getContextPriority(Query<T> query, int defaultValue)
```

Comment:
I see what you mean. We deprecated them because we planned avoid using string parsing going forward, but that may warrant a separate discussion. I'm fine leaving as is.